api/service/stagedsync/sync_status.go

```diff
@@ -75,7 +76,9 @@ func (status *syncStatus) Get(fallback func() SyncCheckResult) SyncCheckResult {
 	defer status.lock.Unlock()
 	if status.expired() {
 		result := fallback()
-		status.update(result)
+		if result.OtherHeight > 0 && result.OtherHeight < uint64(math.MaxUint64) {
```

Comment:
if you expect this is possible, please log otherwise case.

Reply:
getMaxHeight returns either **0**  or **math.MaxUint64**, in case it is not able to query other peers. So, here, first we validate the return value, and if it is valid, we update the status. The last code was updating the status regardless of the validity of the returned value.